pkg/controller/awsprivatelink/cleanup.go

```diff
@@ -136,9 +137,13 @@ func (r *ReconcileAWSPrivateLink) cleanupHostedZone(awsClient awsclient.Client,
 		hzID = cd.Status.Platform.AWS.PrivateLink.HostedZoneID
 	}
 
+	// Figure out the API address for cluster. The error handling is only
+	// needed when the hzID is empty because then we will not be able to
+	// discover the hzID and should exit.
+	apiDomain, err := initialURL(r.Client,
+		client.ObjectKey{Namespace: cd.Namespace, Name: metadata.AdminKubeconfigSecretRef.Name})
```

Comment:
Consider DRYing this with the [other instance](https://github.com/openshift/hive/blob/4dfe9ecb15e1db492177e12ab1d5385b98092850/pkg/controller/awsprivatelink/awsprivatelink_controller.go#L540-L542) . (I found it kind of goofy that we were using the kubeconfig to discover the API URL; if that changes at some point we want to catch all paths at once.)

Reply:
I was literally talking about just this one line and the fact that it's feeding `metadata.AdminKubeconfigSecretRef.Name`, which is the goofy part. But meh.